dart/test/flat_buffers_test.dart

```diff
@@ -664,9 +664,11 @@ class BuilderTest {
     }
     // read and verify
     BufferContext buf = BufferContext.fromBytes(byteList);
-    List<int> items = const Uint8ListReader().read(buf, 0);
-    expect(items, hasLength(5));
-    expect(items, orderedEquals(<int>[1, 2, 3, 4, 0x9A]));
+    for (final lazy in [true, false]) {
+      List<int> items = Uint8ListReader(lazy: lazy).read(buf, 0);
+      expect(items, hasLength(5));
+      expect(items, orderedEquals(<int>[1, 2, 3, 4, 0x9A]));
+    }
```

Comment:
We should probably have some assertions in here about whether the instances are the same or not, right?

Reply:
Sorry, not the same instances or not, but we should be asserting that lazy does the lazy thing and non-lazy does not :)

Reply:
updated the test to verify laziness